### PR TITLE
[Merged by Bors] - feat(Data/Quot): add 2 lemmas

### PR DESCRIPTION
**Pull request by urkud**
* Add `Quot.surjective_lift` and `Quotient.surjective_liftOn'`.
* This is the `mathlib4` version of leanprover-community/mathlib#17699.

### CODE REVIEW

**Comment by urkud** [comment]
Do I need some align commands? What are indentation guidelines for mathlib 4?

File: Mathlib/Data/Quot.lean [hunk 1]
```diff
@@ -95,6 +95,10 @@ theorem liftOn_mk (a : α) (f : α → γ) (h : ∀ a₁ a₂, r a₁ a₂ → f
   rfl
 #align quot.lift_on_mk Quot.liftOn_mk
 
+@[simp] theorem surjective_lift {f : α → γ} (h : ∀ a₁ a₂, r a₁ a₂ → f a₁ = f a₂) :
+  Function.Surjective (lift f h) ↔ Function.Surjective f :=
+⟨fun hf => hf.comp Quot.exists_rep, fun hf y => let ⟨x, hx⟩ := hf y; ⟨Quot.mk _ x, hx⟩⟩
```
Comment (semorrison) [inline Mathlib/Data/Quot.lean:100]:
```suggestion
    Function.Surjective (lift f h) ↔ Function.Surjective f :=
  ⟨fun hf => hf.comp Quot.exists_rep, fun hf y => let ⟨x, hx⟩ := hf y; ⟨Quot.mk _ x, hx⟩⟩
```
Indentation is 4 spaces for continuing the type signature, and two spaces for the body.

File: Mathlib/Data/Quot.lean [hunk 2]
```diff
@@ -542,6 +546,10 @@ protected theorem liftOn'_mk'' (f : α → φ) (h) (x : α) :
     Quotient.liftOn' (@Quotient.mk'' _ s₁ x) f h = f x :=
   rfl
 
+@[simp] lemma surjective_liftOn' {f : α → φ} (h) :
+  Function.Surjective (λ x : Quotient s₁ => x.liftOn' f h) ↔ Function.Surjective f :=
+Quot.surjective_lift _
```
Comment (semorrison) [inline Mathlib/Data/Quot.lean:551]:
```suggestion
@[simp] lemma surjective_liftOn' {f : α → φ} (h) :
  Function.Surjective (λ x : Quotient s₁ => x.liftOn' f h) ↔ Function.Surjective f :=
Quot.surjective_lift _
#align quotient.surjective_lift_on' Quotient.surjective_liftOn'
```

File: Mathlib/Data/Quot.lean (hunk 1 above)
Comment (semorrison) [inline Mathlib/Data/Quot.lean:100]:
```suggestion
@[simp] theorem surjective_lift {f : α → γ} (h : ∀ a₁ a₂, r a₁ a₂ → f a₁ = f a₂) :
  Function.Surjective (lift f h) ↔ Function.Surjective f :=
⟨fun hf => hf.comp Quot.exists_rep, fun hf y => let ⟨x, hx⟩ := hf y; ⟨Quot.mk _ x, hx⟩⟩
#align quot.surjective_lift Quot.surjective_lift
```
But this one probably isn't needed as I think mathport gets it correct.

File: Mathlib/Data/Quot.lean (hunk 2 above)
Comment (urkud) [reply Mathlib/Data/Quot.lean:551]: Should I increase indentation here too?

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**